astropy/coordinates/angles.py

```diff
@@ -587,7 +587,7 @@ def _validate_angles(self, angles=None):
         if angles.unit is u.deg:
             limit = 90
         elif angles.unit is u.rad:
-            limit = 0.5 * np.pi
+            limit = np.asanyarray(0.5 * np.pi, dtype=self.dtype)
```

Comment:
for speed purposes, `limit = self.dtype.type(0.5 * np.pi)` seems fastest (even for unchanged dtype)

Reply:
done